### PR TITLE
changed user UserAgent name

### PR DESCRIPTION
**Pull request by ATREAY**
## Description
This pull request introduces modifications to the user agent/network name.

## Issues
fixes: #2932 , #2908

### CODE REVIEW

**Comment by Wondertan** [comment]
@ATREAY, This works and solves the issue. Let's also fix #2908 in this PR. You can get it from `nodebuilder/node.GetBuildInfo`

**Comment by ATREAY** [comment]
Certainly, @Wondertan.
Since I'm new to the repo, could you please clarify whether the system should return the semantic version or the system version?

**Comment by Wondertan** [comment]
@ATREAY, semantic. System version should stay private

**Comment by ATREAY** [comment]
Hey @Wondertan, I included the network version. Could you confirm if this meets the requirements mentioned in the issue?

File: nodebuilder/p2p/host.go [hunk 1]
```diff
@@ -41,14 +41,16 @@ func routedHost(base HostBase, r routing.PeerRouting) hst.Host {
 
 // host returns constructor for Host.
 func host(params hostParams) (HostBase, error) {
+	buildInfo := node.GetBuildInfo()
+	userAgent := fmt.Sprintf("%s/%s", params.Net, buildInfo.SemanticVersion)
```
Comment (Wondertan) [inline nodebuilder/p2p/host.go:45]: Let's make it so it has a form of `celestia-node/Arabica-10/v0.12/abc32das`

Comment (ATREAY) [reply nodebuilder/p2p/host.go:45]: Hi @Wondertan, I need a little help. How can I get the network alias (like arabica, mocha, etc.) from the network.go file to host.go?

Comment (Wondertan) [reply nodebuilder/p2p/host.go:45]: It's already passed through via params.Net

**Comment by ATREAY** [comment]
Hey @Wondertan, I have formatted the user agent name. Could you please check and confirm if it is correct when you have a moment?

File: nodebuilder/p2p/host.go [hunk 2]
```diff
@@ -41,14 +42,28 @@ func routedHost(base HostBase, r routing.PeerRouting) hst.Host {
...
 func host(params hostParams) (HostBase, error) {
+	buildInfo := node.GetBuildInfo()
+	netAlias := params.Net.String()
+
+	// Split into parts
+	parts := strings.SplitN(netAlias, "-", 2)
+	netType := parts[0]
+	alias := parts[1]
+
+	userAgent := fmt.Sprintf("%s-node/%s/v%s/%s",
```
Comment (Wondertan) [inline nodebuilder/p2p/host.go:53]: celestia-node should stay constant

**Comment by ATREAY** [comment]
Hey @Wondertan, Since params.Net contains comprehensive information about the network name and its alias, I believe this should suffice.

**Comment by Wondertan** [comment]
@ATREAY, No. It must be in the following format `celestia-node/Arabica-10/v0.12/abc2das`, currently it will produce `Arabica-10-node/v0.12/abc2das`. Again, celestia-node should stay constant. 

Also, I made a typo initially, and it should be 7 first chars of commit instead of 8.

Please write a unit test that ensures we follow the requested UserAgent form

**Comment by ATREAY** [comment]
Hey @Wondertan 

I want to inform you that I lack familiarity with unit testing techniques and creating unit tests, as I have not worked on them previously. If feasible, could you please assign this task to someone with experience in unit testing? This would help prevent unnecessary errors that may arise if I were to undertake this task.

I apologize for any inconvenience this may cause.

**Comment by ramin** [comment]
@ATREAY i can help with getting some unit tests in place so we can get this merged

**Comment by ATREAY** [comment]
> @ATREAY i created some tests and refactored code here [a7c7b16](https://github.com/celestiaorg/celestia-node/commit/a7c7b16947e3c425ca15ea7dc09f992d6e45863c)
> 
> have a look, if you like and agree i can push the changes to this branch and we can see about getting this merged

I think it's good. What do you think, @Wondertan? Is it okay?

**Comment by ramin** [comment]
@Wondertan check this with some expanded tests, if you like we can replace this PR with it and get over the line: https://github.com/ATREAY/celestia-node/pull/1

**Comment by Wondertan** [comment]
@ramin, lets use your tested version! We can open your PR directly to node

**Comment by ramin** [comment]
replaced by https://github.com/celestiaorg/celestia-node/pull/3379

**Pull request closed without merging**